src/inmanta/deploy/work.py

```diff
@@ -246,6 +246,17 @@ def task_done(self, agent: str, task: tasks.Task) -> None:
         del self._in_progress[task]
         self._agent_queues[agent].task_done()
 
+    def retrieve_task_in_progress(self, task: tasks.Task) -> Optional[tasks.Task]:
+        """
+        Retrieve the task that is currently stored in the Agent Queue. This task should contain information on why it was
+        scheduled.
+        """
+        actual_search = [e for e in self._in_progress.keys() if e == task]
+        if len(actual_search) == 0:
+            return None
+        else:
+            return actual_search[0]
```

Comment:
linear search is a no-go in this code. All of this code is hyper efficient, this is not